staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/admission.go

```diff
@@ -163,19 +165,16 @@ func (c *celAdmissionPlugin) Validate(
 		return nil
 	}
 
-	deadlined, cancel := context.WithTimeout(ctx, 2*time.Second)
-	defer cancel()
+	if !c.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
```

Comment:
Should we have the following `isPolicyResource(a)` check up before this so it will not waste time on policy resource?

Reply:
great catch!